crates/net/src/types/envelope.rs

```diff
@@ -69,11 +72,109 @@ impl ExecutionPayloadEnvelope {
+        }
+    }
+
+    fn convert_payload_v2(payload: ExecutionPayloadV2) -> L2ExecutionPayload {
+        L2ExecutionPayload {
+            parent_hash: payload.payload_inner.parent_hash,
+            fee_recipient: payload.payload_inner.fee_recipient,
+            state_root: payload.payload_inner.state_root,
+            receipts_root: payload.payload_inner.receipts_root,
+            logs_bloom: payload.payload_inner.logs_bloom,
+            prev_randao: payload.payload_inner.prev_randao,
+            block_number: payload.payload_inner.block_number,
+            gas_limit: payload.payload_inner.gas_limit.into(),
+            gas_used: payload.payload_inner.gas_used.into(),
+            timestamp: payload.payload_inner.timestamp,
+            extra_data: payload.payload_inner.extra_data,
+            base_fee_per_gas: Self::convert_uint128(payload.payload_inner.base_fee_per_gas),
+            block_hash: payload.payload_inner.block_hash,
+            transactions: payload.payload_inner.transactions,
+            deserialized_transactions: Vec::default(),
+            withdrawals: Some(Vec::new()),
+            blob_gas_used: None,
+            excess_blob_gas: None,
+        }
+    }
+
+    fn convert_payload_v3(payload: ExecutionPayloadV3) -> L2ExecutionPayload {
+        L2ExecutionPayload {
+            parent_hash: payload.payload_inner.payload_inner.parent_hash,
+            fee_recipient: payload.payload_inner.payload_inner.fee_recipient,
+            state_root: payload.payload_inner.payload_inner.state_root,
+            receipts_root: payload.payload_inner.payload_inner.receipts_root,
+            logs_bloom: payload.payload_inner.payload_inner.logs_bloom,
+            prev_randao: payload.payload_inner.payload_inner.prev_randao,
+            block_number: payload.payload_inner.payload_inner.block_number,
+            gas_limit: payload.payload_inner.payload_inner.gas_limit.into(),
+            gas_used: payload.payload_inner.payload_inner.gas_used.into(),
+            timestamp: payload.payload_inner.payload_inner.timestamp,
+            extra_data: payload.payload_inner.payload_inner.extra_data,
+            base_fee_per_gas: Self::convert_uint128(
+                payload.payload_inner.payload_inner.base_fee_per_gas,
+            ),
+            block_hash: payload.payload_inner.payload_inner.block_hash,
+            transactions: payload.payload_inner.payload_inner.transactions,
+            deserialized_transactions: Vec::default(),
+            withdrawals: Some(Vec::new()),
+            blob_gas_used: Some(payload.blob_gas_used.into()),
+            excess_blob_gas: Some(payload.excess_blob_gas.into()),
+        }
+    }
+
+    fn convert_uint128(value: alloy::primitives::U256) -> Option<u128> {
+        let bytes = value.to_le_bytes_vec();
+        let bytes: [u8; 16] = bytes.try_into().ok()?;
+        Some(u128::from_le_bytes(bytes))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use alloy::hex;
+
```

Comment:
Let's also add a `V1` encoding test please 🙏

Reply:
> Let's also add a `V1` encoding test please 🙏

will do, because i don't find a similar tests for `V1` in optimism repo, i am looking how to get the serialized hex.

Reply:
Added it.